include/rocksdb/transaction_log.h

```diff
@@ -56,8 +56,25 @@ class LogFile {
 };
 
 struct BatchResult {
-  SequenceNumber sequence = 0;
-  std::unique_ptr<WriteBatch> writeBatchPtr;
+    SequenceNumber sequence = 0;
+    std::unique_ptr<WriteBatch> writeBatchPtr;
+
+    BatchResult() {
+    }
+
+    BatchResult(const BatchResult&) = delete;
+
+    BatchResult& operator=(const BatchResult&) = delete;
```

Comment:
Rule of five requires us to declare either all of {copy, copy-assign, move, move-assign, destructor}, or none of them. See here: http://en.cppreference.com/w/cpp/language/rule_of_three